roles/calico/README.rst

```diff
@@ -0,0 +1,147 @@
+Calico
+======
+
+.. versionadded:: 0.4
+
+`Calico <http://www.projectcalico.org>`_ is used in the project to add the IP
+per container functionality. Calico connects Docker containers through IP no matter
+which worker node they are on. Calico uses :doc:`etcd` to distribute information
+about workloads, endpoints, and policy to each worker node. Endpoints are
+network interfaces associated with workloads. Calico is deployed in the Docker
+container on each worker node and managed by systemd. Any workload managed by
+Calico is registered as a service in Consul.
+
+Modes
+^^^^^
+
+Calico can run a public cloud environment that does not allow either L3 peering
+or L2 connectivity between Calico hosts. Calico will then route traffic between
+the Calico hosts using IP in IP mode. At this time, the full node-to-node BGP
+mesh is supported and configured in CiscoCloud only. Other cloud environments
+are set up with the IP in IP mode.
+
+Mesos
+^^^^^
+
+The systemd unit file for Mesos slave is updated with the new ``DOCKER_HOST``
+environment variable:
+
+.. code-block:: shell
+
+   DOCKER_HOST=localhost:2377
+
+This allows Calico to set up networking automatically by routing Docker API
+requests through the `Powerstrip <https://github.com/clusterhq/powerstrip>`_
+proxy that is running on port ``2377`` on each Mesos slave host. It sets
+a dependency, meaning if Calico Docker container is not running, the Mesos slave
+will refuse to start accordingly.
+
+Marathon
+^^^^^^^^
+
+When you start containers on top of Marathon, you will need to add two
+environment variables to your JSON file: ``CALICO_IP`` and ``CALICO_PROFILE``.
+You can assign an IP address to ``CALICO_IP`` explicitly or set ``auto`` and it
+will be allocated automatically. If the profile set with ``CALICO_PROFILE`` doesn't
+exist, it will be created automatically. If you don't provide the two variables,
+the Docker default network settings will be applied.
+
+Example:
+
+.. code-block:: json
+
+   {
+     "container": {
+       "type": "DOCKER",
+       "docker": {
+         "image": "busybox",
+         "parameters":[{"key": "env", "value": "CALICO_IP=auto"},
+                       {"key": "env", "value": "CALICO_PROFILE=dev"}]
```

Comment:
Yes, but the Marathon environment variables are added as command flags automatically to Docker. This is what people will be expecting to do, so please verify if it does or does not work in the docs.